javascript/javascript2/week2/Hyfbay/main.js

```diff
@@ -0,0 +1,99 @@
+console.log('Script loaded');
+const products = getAvailableProducts();
+
+/**********************JS2 - W1 HOMEWORK******************* */
+/*function renderProducts(products) {
+  const ul = document.querySelector('ul');
+  products.forEach((product) => {
+    const li = document.createElement('li');
+    li.innerHTML = `<h2>${product.name}</h2><span>price: ${product.price}</span><br><span>Rating: ${product.rating}</span>`;
+    ul.appendChild(li);
+  });
+}
+
+/************************* JS - W2 HOMEWORK *************************** */
+const source = document.getElementById('input-text');
+const ul = document.querySelector('ul');
+
+function quickSearch() {
+  const filterSearchProduct = document.getElementById('input-text').value;
+
+  products.filter((product) => {
```

Comment:
I believe you need to do more research on `map, reduce and filter`. The code bellow is not correct. See some simple explanations here. https://levelup.gitconnected.com/sneak-peak-of-map-filter-and-reduce-in-javascript-79d38181a48 . Also try to do some exercises with this concept here:https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/functional-programming/use-higher-order-functions-map-filter-or-reduce-to-solve-a-complex-problem